python/lsst/pipe/drivers/coaddDriver.py

```diff
@@ -0,0 +1,297 @@
+        with self.logOperation("warping %s" % (patchRef.dataId,), catch=True):
+            self.makeCoaddTempExp.run(patchRef, selectDataList)
+        return selectDataList
+
+    def coadd(self, cache, data):
+        """Construct coadd for a patch and measure
+        Only slave nodes execute this method.
+        Because only one argument may be passed, it is expected to
+        contain multiple elements, which are:
+        @param patchRef: data reference for patch
+        @param selectDataList: List of SelectStruct for inputs
+        """
+        patchRef = getDataRef(cache.butler, data.patchId, cache.coaddType)
+        selectDataList = data.selectDataList
+        coadd = None
+        with self.logOperation("coadding %s" % (patchRef.dataId,), catch=True):
+            if self.config.doOverwriteCoadd or not patchRef.datasetExists(cache.coaddType):
+                coaddResults = self.assembleCoadd.run(patchRef, selectDataList)
+                if coaddResults is not None:
+                    coadd = coaddResults.coaddExposure
+            elif patchRef.datasetExists(cache.coaddType):
+                self.log.info("%s: Reading coadd %s" % (NODE, patchRef.dataId))
+                coadd = patchRef.get(cache.coaddType, immediate=True)
+
+        if coadd is None:
+            return
+
+        with self.logOperation("detection on %s" % (patchRef.dataId,), catch=True):
+            idFactory = self.detectCoaddSources.makeIdFactory(patchRef)
+            # This includes background subtraction, so do it before writing the coadd
+            detResults = self.detectCoaddSources.runDetection(coadd, idFactory)
+            self.detectCoaddSources.write(coadd, detResults, patchRef)
+
+        self.assembleCoadd.writeCoaddOutput(patchRef, coadd, "calexp")
+
+
+    def selectExposures(self, patchRef, selectDataList):
+        """Select exposures to operate upon, via the SelectImagesTask
+        This is very similar to CoaddBaseTask.selectExposures, except we return
+        a list of SelectStruct (same as the input), so we can plug the results into
+        future uses of SelectImagesTask.
+        """
+        key = lambda dataRef: tuple(dataRef.dataId[k] for k in sorted(dataRef.dataId.keys()))
+        inputs = dict((key(select.dataRef), select) for select in selectDataList)
+        skyMap = patchRef.get(self.config.coaddName + "Coadd_skyMap")
+        tract = skyMap[patchRef.dataId["tract"]]
+        patch = tract[(tuple(int(i) for i in patchRef.dataId["patch"].split(",")))]
+        bbox = patch.getOuterBBox()
+        wcs = tract.getWcs()
+        cornerPosList = afwGeom.Box2D(bbox).getCorners()
+        coordList = [wcs.pixelToSky(pos) for pos in cornerPosList]
+        dataRefList = self.select.runDataRef(patchRef, coordList, selectDataList=selectDataList).dataRefList
+        return [inputs[key(dataRef)] for dataRef in dataRefList]
+
+    def writeMetadata(self, dataRef):
+        pass
```

Comment:
`writeMetadata` is an action rather than an accessor, so `pass` seems appropriate.